bundles/org.openhab.binding.electroluxappliances/src/main/java/org/openhab/binding/electroluxappliances/internal/handler/ElectroluxAppliancesHandlerFactory.java

```diff
@@ -0,0 +1,132 @@
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.util.Set;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.HttpProxy;
+import org.eclipse.jetty.client.ProxyConfiguration;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link ElectroluxAppliancesHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.electroluxappliances", service = ThingHandlerFactory.class)
+public class ElectroluxAppliancesHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_ELECTROLUX_AIR_PURIFIER,
+            THING_TYPE_ELECTROLUX_WASHING_MACHINE, THING_TYPE_BRIDGE);
+    private final Gson gson;
+    private HttpClient httpClient;
+    private final Logger logger = LoggerFactory.getLogger(ElectroluxAppliancesHandlerFactory.class);
+    private static final boolean DEBUG = false;
+
+    @Activate
+    public ElectroluxAppliancesHandlerFactory(@Reference HttpClientFactory httpClientFactory) {
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+        this.gson = new Gson();
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (THING_TYPE_ELECTROLUX_AIR_PURIFIER.equals(thingTypeUID)) {
+            return new ElectroluxAirPurifierHandler(thing);
+        } else if (THING_TYPE_ELECTROLUX_WASHING_MACHINE.equals(thingTypeUID)) {
+            return new ElectroluxWashingMachineHandler(thing);
+        } else if (THING_TYPE_BRIDGE.equals(thingTypeUID)) {
+            return new ElectroluxAppliancesBridgeHandler((Bridge) thing, httpClient, gson);
+        }
+        return null;
+    }
+
+    @Reference
+    protected void setHttpClientFactory(HttpClientFactory httpClientFactory) {
```

Comment:
This seems some testing / debugging code. Please remove or extract to a unit test class

Reply:
Sorry, forgot to remove that, fixed!

Reply:
Its still there.

Reply:
My bad! Fixed!